README.md

```diff
@@ -1,132 +1,488 @@
+
+Following are some issues/errors that one may encounter while running the pipeline, and the ways you can avoid or recover from them.
+
+* Duplicate rows: When writing data to a BigQuery table, Cloud DLP might write duplicate rows.
+
+    The project uses the Streaming Inserts API of BigQuery which by default enables best-effort deduplication mechanism. For a possible solution, see the knowledge base article, checkout [high number of duplicates in Dataflow pipeline streaming inserts to BigQuery](https://cloud.google.com/knowledge/kb/high-number-of-duplicates-in-dataflow-pipeline-streaming-inserts-to-bigquery-000004276?authuser=0).
+
+
+* Errors in DLPTransform step: 
+
+  (The detailed errors can be viewed in worker logs for the mentioned PTransform.)
+   
+    1. ```INVALID_ARGUMENT: Too many findings to de-identify. Retry with a smaller request.```
+
+       DLP has a max findings per request [limit](https://cloud.google.com/dlp/limits#content-redaction-limits) of 3000. Run the pipeline again with a smaller batch size.
+       
+    2. ```RESOURCE_EXHAUSTED: Quota exceeded for quota metric 'Number of requests' and limit 'Number of requests per minute' of service 'dlp.googleapis.com'```
+
+       This can happen if the Dataflow pipeline is being run with a small batch size. Re-run the pipeline with a larger batch size.
+
+       If increasing the batch size is not possible or if the issue persists even after reaching the maximum batch size, consider trying one of the following options:
+
+       * The pipeline incorporates a retry mechanism for DLP API calls, utilizing an exponential delay approach. To enhance the retry behavior, you can adjust the value of the `dlpApiRetryCount` parameter. For more information, please refer to the [pipeline parameters](#pipeline-parameters) section, specifically the `dlpApiRetryCount` and `initialBackoff` parameters.
+
+       * The pipeline includes a parameter called `numShardsPerDLPRequestBatching`. Decreasing this value below the default (100) will result in a lower number of concurrent requests sent to DLP.
+
+       * Verify if there are any other pipelines or clients generating DLP API requests.
+
+       * Consider submitting a request to increase the quota limit. Refer [increase dlp quota](https://cloud.google.com/dlp/limits#increases).
+
+
+## Dataflow DAG
+
+For inspection and de-identification:
+![v2_dag_](diagrams/dlp_dag_new.png)
+
+
+
+For re-identification
+
+![v2_dag_](diagrams/dlp_reid_dag.png)
+
+
+
+## Advanced topics
+* Dataflow templates allow you to package a Dataflow pipeline for deployment. Instead of having to build the pipeline everytime, you can create flex templates and deploy the template by using the Google Cloud console, the Google Cloud CLI, or REST API calls.
+* For more details, refer [Dataflow templates](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates) and [flex templates](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates).
+
```

Comment:
nit: Dataflow documentation consistently uses "Flex Templates". We should align with that.

Reply:
Updated, thanks.